sdk/iOS/src/MSSyncContext.h

```diff
@@ -54,19 +54,19 @@
 /// @{
 
 /// Returns a dictionary containing the items and totalCount
-- (MSSyncContextReadResult *) readWithQuery:(MSQuery *)query orError:(NSError **)error;
+- (nullable MSSyncContextReadResult *) readWithQuery:(nullable MSQuery *)query orError:(NSError * __nullable * __nullable)error;
 
 /// Should retrieve a single item from the local store or nil if item with the given ID does not exist.
--(NSDictionary *) readTable:(NSString *)table withItemId:(NSString *)itemId orError:(NSError **)error;
+-(nullable NSDictionary *) readTable:(nonnull NSString *)table withItemId:(nonnull NSString *)itemId orError:(NSError * __nullable * __nullable)error;
 
 /// Should insert/update the given item in the local store as appropriate
--(BOOL) upsertItems:(NSArray *)item table:(NSString *)table orError:(NSError **)error;
+-(BOOL) upsertItems:(nullable NSArray *)item table:(nonnull NSString *)table orError:(NSError * __nullable * __nullable)error;
 
 /// Should remove the provided item from the local store
--(BOOL) deleteItemsWithIds:(NSArray *)items table:(NSString *)table orError:(NSError **)error;
+-(BOOL) deleteItemsWithIds:(nullable NSArray *)items table:(nonnull NSString *)table orError:(NSError * __nullable * __nullable)error;
 
 /// Should remove all entries from the specified table in the local store
--(BOOL) deleteUsingQuery:(MSQuery *)query orError:(NSError **)error;
+-(BOOL) deleteUsingQuery:(nullable MSQuery *)query orError:(NSError * __nullable * __nullable)error;
```

Comment:
I guess it will work, but not the intent.  Since any 3rd party could conform to this, I think it feels cleaner to mark it as non null so there is not differing behavior long term if its passed as nil.